docs/operator-guide.md

```diff
@@ -0,0 +1,289 @@
+# UFDS Credential Rotation
+
+UFDS key rotation impacts multiple feature areas and requires about an hour to complete from start to finish.
+
+The functional areas impacted during the change window include:
+
+- authentication for Triton, Manta API, and the Triton Service portal
+- smartlogin
+- account CRUD and fabric creation
+- firewall rule CRUD
+
+## Procedure summary
+
+The high-level steps are as follows:
+
+1. Update the data centers that have UFDS replicas by:
+    - Setting a new, local UFDS password.
+    - Configuring the passwords so they still talk to the UFDS primary using the current password. During this stage, only the replica data center being updated is impacted.
+
+2. Update the data center containing the UFDS primary by:
+    - Setting a new, local UFDS password. Once this is done, all services connecting to the UFDS primary will no longer work. These services include, but is not limited to, all `ufds-replicators`, portal, and cloudapi/adminui requests that write to UFDS.
+    - Updating the remote UFDS password on each data center that has UFDS replicas.
+
+### Before you begin
+
+- **Determine whether you have enabled UFDS replication**. Parts of this guide assume that you have already enabled UFDS replication using the `sdc-ufds-m2s` script. If you have not set up UFDS replication, follow the steps for the UFDS primary only. Skip all sections that refer to replicas.
+
+- **Generate passwords**. You may use a single password for all data centers because they have the same content. You may also use a unique password per data center. If you use unique passwords, then all replica data centers must use the primary's password as the remote password.
+
+    If you are unsure which data center is the primary, check `/usbkey/config` on the headnode for the line `ufds_is_master=true`. There should be only *one* datacenter with this setting. If for some reason you have more than one data center with `ufds_is_master=true` and are unsure how to proceed, contact Triton support.
+
+- **Prepare to monitor UFDS log files**. Monitor the logs while making the changes is a best practice to catch anything that might go wrong.
+
+    Each of these commands are run from the UFDS zone.
+
+    ```sh
+    sdc-login -l ufds
+    ```
+
+  - To monitor the UFDS application log (all datacenters):
+
+    ```sh
+    tail -1f $(svcs -L ufds-master) | bunyan
+    ```
+
+  - To monitor the replication log (replicas only):
+
+    ```sh
+    tail -1f $(svcs -L ufds-replicator) | bunyan
+    ```
+
+## Procedure
+
+### Start data center maintenance
+
+These commands are run from the headnode global zone.
+
+```sh
+sdcadm dc-maint start --message="This DC is in maintenance."
+
+# confirm status
+sdcadm dc-maint status
+```
+
+### On each data center that has a UFDS replica
+
+1. Run `sdc-usbkey mount`.
+
+2. Update `/usbkey/config` and `/mnt/usbkey/config` to set `ufds_ldap_root_pw` to the new `<REPLICA_PW>`.
+
```

Comment:
If I'm a CN without a USB key, then what?  A HN will have its "USB key" on $BOOTPOOL, but a CN that does, say, PXE/EFI-network-to-iPXE booting, then what?

Reply:
In a usb-less environment, `sdc-usbkey mount` will mount `$BOOTPOOL` on `/mnt/usbkey`. So the instructions will still be valid.